com.onesignal.unity.core/Runtime/User/Models/IUserState.cs

```diff
@@ -0,0 +1,46 @@
+/*
+ * Modified MIT License
+ *
+ * Copyright 2023 OneSignal
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * 1. The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * 2. All copies of substantial portions of the Software may only be used in connection
+ * with services provided by OneSignal.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+using System;
+
+namespace OneSignalSDK.User {
+    public interface IUserState {
+        /// <summary>
+        /// The unique identifier for your OneSignal account. This will be an empty string until the
+        /// user has been successfully logged in on the backend and assigned an ID.
+        /// Use [addObserver] to be notified when the [OnesignalId] has been successfully assigned.
+        /// </summary>
+        string OneSignalId { get; }
+        
+        /// <summary>
+        /// The external identifier that you use to identify users. Use [addObserver] to be notified
+        /// when the [externalId] has been successfully assigned. This will be an empty string if no
```

Comment:
Same as above for addObserver to Changed Event Handler

Also update [externalId] to ExternalId to match property